src/oemof/solph/flows/_flow.py

```diff
@@ -250,6 +255,29 @@ def __init__(self, **kwargs):
                 + "nonconvex flows!"
             )
 
+        infinite_error_msg = (
+            "{} must be a finite value. Passing an infinite "
+            "value is not allowed."
+        )
+        if not self.investment:
+            warn_msg_no_nominal_value = (
+                "If {} is set in a flow (except InvestmentFlow), "
+                "nominal_value must be set as well.\n"
+                "Otherwise, it won't have any effect."
+            )
+            if self.nominal_value is None:
+                for attr in need_nominal_value:
+                    if kwargs.get(attr) is not None:
+                        warn(
+                            warn_msg_no_nominal_value.format(attr),
+                            debugging.SuspiciousUsageWarning,
+                        )
+            else:
+                assert math.isfinite(
+                    self.nominal_value
+                ), infinite_error_msg.format("nominal_value")
+        assert math.isfinite(self.max[0]), infinite_error_msg.format("max")
```

Comment:
Why is only max checked for a finite value and not min/full_load_time_max....
Why do you provoke an `AssertationError`, for me it is a `ValueError`, but I am not sure.